scripts/build_datapackage.py

```diff
@@ -260,80 +378,190 @@ def frame_to_schema(frame: pl.LazyFrame | pl.DataFrame, /) -> fl.Schema:
+                f"Expected only lists of mappings or single values, "
+                f"but got:{type(item).__name__!r}\n{item!r}\n\n{extra!r}"
+            )
+            raise NotImplementedError(msg)
+
+    # Remaining are in-spec and only in `datapackage_additions.toml`
+    changes |= additions
+    return PackageMeta(**changes)
+
+
+def extract_package_metadata(
+    npm: Mapping[str, Any], sources: Mapping[str, Any], /
+) -> PackageMeta:
+    pkg_meta = _extract_npm_metadata(npm)
+    return _merge_package_metadata(pkg_meta, sources)
+
+
+def extract_overrides(resources: Any, /) -> dict[str, ResourceMeta]:
+    if isinstance(resources, Sequence):
+        return dict(iter_parse_resources(resources))
+    msg = (
+        f"Expected `resources` to be an array of tables, but got:"
+        f"\n{type(resources).__name__!r}\n\n{resources!r}"
+    )
+    raise TypeError(msg)
+
+
+def iter_parse_resources(
+    seq: Sequence[ResourceMeta], /
+) -> Iterator[tuple[str, ResourceMeta]]:
+    for resource in seq:
+        if (name := resource.get("path")) and isinstance(name, str):
+            m: Any = {k: v for k, v in resource.items() if k != "path"}
+            # NOTE: Drops entries that only provide `path`
+            if m:
+                yield name, m
         else:
+            raise TypeError(seq)
+
+
+def iter_data_dir(data_root: Path, /) -> Iterator[Path]:
+    """Yield files in the root of the ``/data/`` directory."""
+    for fp in sorted(data_root.iterdir()):
+        if fp.is_file():
+            yield fp
+
+
+def iter_resources(
+    root: Path, /, overrides: dict[str, ResourceMeta]
+) -> Iterator[Resource]:
+    """
+    Yield all parseable resources, constructing with the most appropriate ``Resource`` class.
+
+    Parameters
+    ----------
+    root
+        Directory storing datasets.
+    overrides
+        Additional metadata, with a higher precedence than inferred.
+    """
+    for fp in iter_data_dir(root):
+        if not ResourceAdapter.is_supported(fp):
             msg = f"Skipping unexpected extension {fp.suffix!r}\n\n{fp!r}"
             warnings.warn(msg, stacklevel=2)
             continue
+        resource = ResourceAdapter.from_path(fp)
+        name = fp.name
+        if name in overrides:
+            resource = ResourceAdapter.with_extras(resource, **overrides[name])
+        yield resource
+
+
+def read_toml(fp: Path, /) -> dict[str, Any]:
+    return tomllib.loads(fp.read_text("utf-8"))
+
+
+def read_json(fp: Path, /) -> Any:
+    with fp.open(encoding="utf-8") as f:
+        return json.load(f)
+
+
+def write_package(pkg: Package, repo_dir: Path, *formats: OutputFormat) -> None:
+    """Write the final datapackage in one or more formats."""
+    configs: dict[OutputFormat, tuple[str, PackageMethod[str]]] = {
+        "json": (".json", partial(Package.to_json)),
+        "yaml": (".yaml", partial(Package.to_yaml)),
+        "md": ("-tabular.md", partial(Package.to_markdown, table=True)),
```

Comment:
### FIXME
- [ ] Replace `"-tabular.md"` -> `".md"`
- [ ] Re-run `build_datapackage.py`

Reply:
> I'm thinking we can refine the templates further (if anyone has ideas).

Looking very good. I'm sure we can refine the template style in future PRs. My one initial thought is whether the markdown would benefit from a table of contents at the top to allow easier navigation? 

> Then rename the file to something other than datapackage-tabular

I'm wondering if there's a good enough reason to call this file something other than SOURCES.md, given that is what users of this repo have come to expect.

Reply:
> Looking very good. I'm sure we can refine the template style in future PRs.

@dsmedia sounds good to me.

> My one initial thought is whether the markdown would benefit from a table of contents at the top to allow easier navigation?

I haven't found a way to set the view in the url, but viewing on GitHub already provides an *Outline* option for this:

https://github.com/user-attachments/assets/7854cb03-7cf5-4232-b878-2c7f3dd94ae6

Another option could be using the markdown as a source for GitHub pages, eg.
- https://github.com/vega/vega/blob/main/docs/docs/expressions.md
- https://vega.github.io/vega/docs/expressions/

Reply:
> @dangotbanned
> > Then rename the file to something other than datapackage-tabular
>  
> I'm wondering if there's a good enough reason to call this file something other than `SOURCES.md`, given that is what users of this repo have come to expect.

@dsmedia I dont feel *super* strongly either way, but to me it seems simplest to have a shared name for representations of the same data:
- `datapackage.json`
- `datapackage.md`

It wouldn't be clear to me that there was a relationship between these two:
- `datapackage.json`
- `SOURCES.md`

Reply:
> Another option could be using the markdown as a source for GitHub pages, eg.
> 
> * https://github.com/vega/vega/blob/main/docs/docs/expressions.md
> * https://vega.github.io/vega/docs/expressions/

Interesting idea. As this isn't core to the move to Frictionless datapackage generation, I'm happy to leave the discussion around improving markdown navigation for a separate issue.

> It wouldn't be clear to me that there was a relationship between these two:

Good point. It does seem more important to make it clear that the `json` and `md` files are related.

Reply:
Fixed in (https://github.com/vega/vega-datasets/pull/643/commits/c97ad7d156f35cfcc060fd00700012a5dd3cdf60)